### PR TITLE
Fix multiple aggregation edge cases

### PR DESCRIPTION
**Pull request by dirkkul**
This PR contains fixes + tests for 5 different edge cases where aggregation would fail. Each fix and the corresponding tests are in one commit and can be reviewed commit by commit.

1. Fix aggregation with dates as string
2. Fix filtered aggegration with arrays
3. Fix aggregation with empty/non-existent text properties (fixes https://github.com/semi-technologies/weaviate/issues/1675)
4. Fix aggregate for objects with empty properties
5. Fix aggregation for classes without properties

Chaos pipeline passed: https://github.com/semi-technologies/weaviate-chaos-engineering/runs/8086646367

### CODE REVIEW

File: adapters/repos/db/aggregator/filtered.go [hunk 1]
```diff
@@ -135,38 +135,111 @@ func (fa *filteredAggregator) analyzeObject(ctx context.Context,
...
 		}
-		if err := prop.boolAgg.AddBool(asBool); err != nil {
-			return err
+		switch prop.dataType {
+		case schema.DataTypeBoolean:
+			if err := analyzeBool(value); err != nil {
+				return err
+			}
+		case schema.DataTypeBooleanArray:
+			valueStruct := value.([]interface{})
```
Comment (antas-marcin) [inline adapters/repos/db/aggregator/filtered.go:154]: here you can also double check if the casting worked properly like so:
```
valueStruct, ok := value.([]interface{})
if !ok {
	return some error
}
```

File: adapters/repos/db/aggregator/filtered.go [hunk 2]
```diff
@@ -135,38 +135,111 @@ func (fa *filteredAggregator) analyzeObject(ctx context.Context,
...
 		}
-		if err := prop.textAgg.AddText(asString); err != nil {
-			return err
+		switch prop.dataType {
+		case schema.DataTypeText, schema.DataTypeString:
+			if err := analyzeString(value); err != nil {
+				return err
+			}
+		case schema.DataTypeTextArray, schema.DataTypeStringArray:
+			valueStruct := value.([]interface{})
```
Comment (antas-marcin) [inline adapters/repos/db/aggregator/filtered.go:206]: here also:
```
valueStruct, ok := value.([]interface{})
if !ok {
	return some error
}
```

File: adapters/repos/db/aggregator/filtered.go [hunk 3]
```diff
@@ -135,38 +135,111 @@ func (fa *filteredAggregator) analyzeObject(ctx context.Context,
...
 		}
-		if err := prop.dateAgg.AddTimestamp(asString); err != nil {
-			return err
+		switch prop.dataType {
+		case schema.DataTypeDate:
+			if err := analyzeDate(value); err != nil {
+				return err
+			}
+		case schema.DataTypeDateArray:
+			valueStruct := value.([]interface{})
```
Comment (antas-marcin) [inline adapters/repos/db/aggregator/filtered.go:232]: a here:
```
valueStruct, ok := value.([]interface{})
if !ok {
	return some error
}
```

File: adapters/repos/db/sorter/lsm_value_extractor.go [hunk 4]
```diff
@@ -34,6 +34,11 @@ func newPropertyExtractor(className schema.ClassName,
 
 func (e *lsmPropertyExtractor) getProperty(v []byte) interface{} {
 	prop, success, _ := storobj.ParseAndExtractProperty(v, e.property)
+
```
Comment (antas-marcin) [inline adapters/repos/db/sorter/lsm_value_extractor.go:37]: you can remove this empty line

**Review by antas-marcin — APPROVED** [review]

File: test/acceptance/graphql_resolvers/setup_test.go [hunk 5]
```diff
@@ -937,12 +977,22 @@ func addTestDataArrayClasses(t *testing.T) {
...
 				"2022-06-01T22:18:59.640162Z",
 			},
+			"dates": []time.Time{
+				time.Date(467, 6, 7, 8, 9, 10, 11, time.UTC),
+			},
 		},
 	})
 	assertGetObjectEventually(t, arrayClassID3)
+
+	// objects with null objects
```
Comment (aliszka) [inline test/acceptance/graphql_resolvers/setup_test.go:990]: objects with null properties?

**Review by aliszka — APPROVED** [review]

File: adapters/repos/db/aggregator/filtered.go (hunk 1 above)
Comment (dirkkul) [reply adapters/repos/db/aggregator/filtered.go]: added, thanks!

File: test/acceptance/graphql_resolvers/setup_test.go (hunk 5 above)
Comment (dirkkul) [reply test/acceptance/graphql_resolvers/setup_test.go]: yes, changed!

File: adapters/repos/db/aggregator/filtered.go [hunk 6]
```diff
@@ -135,38 +135,120 @@ func (fa *filteredAggregator) analyzeObject(ctx context.Context,
...
 		}
-		if err := prop.textAgg.AddText(asString); err != nil {
-			return err
+		switch prop.dataType {
+		case schema.DataTypeText, schema.DataTypeString:
+			if err := analyzeString(value); err != nil {
+				return err
+			}
+		case schema.DataTypeTextArray, schema.DataTypeStringArray:
+			valueStruct := value.([]interface{})
```
Comment (antas-marcin) [inline adapters/repos/db/aggregator/filtered.go:212]: missed this one

Comment (dirkkul) [reply adapters/repos/db/aggregator/filtered.go]: now, thx!

**Review by antas-marcin — APPROVED** [review]
